Userland/Libraries/LibJS/Runtime/TypedArray.cpp

```diff
@@ -199,8 +258,15 @@ void TypedArrayBase::visit_edges(Visitor& visitor)
                 if (vm.exception())                                                                                                    \
                     return {};                                                                                                         \
             } else {                                                                                                                   \
-                /* FIXME: Initialize from Iterator or Array-like object */                                                             \
-                TODO();                                                                                                                \
+                auto iterator = first_argument.as_object().get(vm.well_known_symbol_iterator(), {}, true);                             \
+                if (iterator.is_function()) {                                                                                          \
+                    auto values = iterable_to_list(global_object(), first_argument, iterator);                                         \
+                    initialize_typed_array_from_list(global_object(), *typed_array, values);                                           \
```

Comment:
Missing exception check, I'd suggest moving the one below just outside the if/else